localstack/services/kinesis/resource_providers/aws_kinesis_stream.py

```diff
@@ -138,14 +138,24 @@ def delete(
           - kinesis:DeleteStream
           - kinesis:RemoveTagsFromStream
         """
-        model = request.desired_state
-        request.aws_client_factory.kinesis.delete_stream(
-            StreamARN=model["Arn"], EnforceConsumerDeletion=True
-        )
-        return ProgressEvent(
-            status=OperationStatus.SUCCESS,
-            resource_model={},
-        )
+        model = request.previous_state
+        client = request.aws_client_factory.kinesis
+
+        if not request.custom_context.get(REPEATED_INVOCATION):
+            client.delete_stream(StreamARN=model["Arn"], EnforceConsumerDeletion=True)
+            request.custom_context[REPEATED_INVOCATION] = True
+
+        try:
+            client.describe_stream(StreamARN=model["Arn"])
+            return ProgressEvent(
```

Comment:
All good with this changes for Kinesis.
We should probably check in future, if we are missing clean up for any other service